lang/cs/Org.Apache.REEF.Wake/Remote/IType.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement to express their type
+    /// </summary>
+    public interface IType
+    {
+        /// <summary>
+        /// Returns the type of the class
+        /// </summary>
+        Type ClassType { get; set; }
```

Comment:
in DefaultRemoteManager Transport Layer defines T to be IRemoteEvent and we
cannot instantiate it by using typeof(IRemoteEvent). Thats why.

Dhruv

On Thu, Apr 16, 2015 at 4:13 PM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Wake/Remote/IType.cs
> <https://github.com/apache/incubator-reef/pull/149#discussion_r28558486>:
>
> > +using System.Collections.Generic;
> > +using System.Linq;
> > +using System.Text;
> > +using System.Threading.Tasks;
> > +
> > +namespace Org.Apache.REEF.Wake.Remote
> > +{
> > +    /// <summary>
> > +    /// Interface that classes should implement to express their type
> > +    /// </summary>
> > +    public interface IType
> > +    {
> > +        /// <summary>
> > +        /// Returns the type of the class
> > +        /// </summary>
> > +        Type ClassType { get; set; }
>
> why not just use typeof()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/149/files#r28558486>.
>